provider/ec2/environ.go

```diff
@@ -66,11 +67,21 @@ var (
 	_ environs.FirewallFeatureQuerier = (*environ)(nil)
 )
 
+type ec2Client interface {
```

Comment:
I've switched from using the full `ec2iface.EC2API` (which has ~1100 methods!) to this much smaller interface of just the stuff we currently need. We used to mock just the methods we needed using interface embedding in `mock_test.go`, but that causes a runtime nil panic if you miss something, whereas this is compile-time type safe. So this approach is better and clearer IMO.

Reply:
Can we do this as well to ensure there's no typos in ec2Client

```
  var _ ec2Client = (ec2iface.EC2API)(nil)
```